tests/lit-tests/signed-integer-overflow.ispc

```diff
@@ -0,0 +1,52 @@
+// This test checks for the use of the no signed wrap (`nsw`) attribute for arithmetic
+// expressions which may suffer from overflow.
+// Needed for induction variable simplification dicussed in
+// [GitHub issue ISPC/#2460](https://github.com/ispc/ispc/issues/2460).
+
+// RUN: %{ispc} %s --dump-file --debug-phase=first --nostdlib --emit-llvm-text --target=host -o /dev/null 
+// RUN: FileCheck %s --input-file ir_0_VerifierPass.ll --check-prefix=CHECK_NSW
+
+// CHECK_NSW-LABEL:         define {{.*}} @mul_u
+// CHECK_NSW-COUNT-2:         {{.*}}= mul{{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @mul_v
+// CHECK_NSW-COUNT-2:         {{.*}}= mul{{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @add_u
+// CHECK_NSW-COUNT-2:         {{.*}}= add{{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @add_v
+// CHECK_NSW-COUNT-2:         {{.*}}= add{{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @sub_u
+// CHECK_NSW-COUNT-2:         {{.*}}= sub{{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @sub_v
+// CHECK_NSW-COUNT-2:         {{.*}}= sub{{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @shl_u
+// CHECK_NSW-COUNT-1:         {{.*}}= shl{{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @shl_v
+// CHECK_NSW-COUNT-1:         {{.*}}= shl{{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @neg_u
+// CHECK_NSW-COUNT-1:         {{.*}}= sub{{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @neg_v
+// CHECK_NSW-COUNT-1:         {{.*}}= sub{{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @ptrMath
+// CHECK_NSW-NOT:             nsw
+
+unmasked uniform int mul_u(uniform int a, uniform int b, uniform int k) { return a * k * b; }
```

Comment:
What's the point checking two multiplies with the same properties/types?

Instead multiplies of different types should be checked. int8/int16/int32/int64.

Reply:
Why is this marked as resolved?

Reply:
I added some other types. Do you want to have one uniform and one varying version for each type?

Reply:
1. int8 is not covered at all.
2. mixing types is not good idea for this matter - the operation is done in the larger type, so only larger type is tested.

Reply:
The code is not very sensitive to uniform/varying. I'm fine is half of the types are tested uniform and half varying.